### PR TITLE
Add Kurdish language

### LINKED ISSUE #17600: Add Kurdish as a new supported language
**Is your feature request related to a problem?**
It is not a bug, but a default parameter to update.

**Describe the solution you'd like**
For the 1.7.8 version, following recent activity on Crowdin, it would be nice to add Kurdish (`ku-IQ`) as a new supported language in PrestaShop.

**Why?**
Some languages are available in Crowdin for translation but not available in PrestaShop because of a low completion level on Crowdin. But when translators are very active and/or showed interest in an unavailable language, it could be nice to add it to the software so that they can see the results of their Crowdin translations quickly in their own shop, and therefore continue translation work.

**How?**
Here is the process I know: to activate a language and have it available in the language drop-down, you need to add it to files _[all_languages.json](https://github.com/PrestaShop/PrestaShop/blob/develop/app/Resources/all_languages.json)_ and _[app/Resources/legacy-to-standard-locales.json](https://github.com/PrestaShop/PrestaShop/blob/develop/app/Resources/legacy-to-standard-locales.json)_.

**Here is the main information about it:**

Kurdish (کوردی)
ku-IQ (ku being the language code for the Kurdish and IQ the country code for Iraq)

Note that the [TranslationFiles repository](https://github.com/PrestaShop/TranslationFiles/blob/master/1.7/translations/1.7.7/available_languages.json) must be up-to-date!

Once the language is officially available (after the release of the version adding it), I will have to update the Locale table in Sandrine.

### PR DESCRIPTION
**Pull request by SalarNazar**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add the support for Kurdish language
| Type?         | improvement
| Category?     | LO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #17600
| How to test?  |0. Open `app/AppKernel.php` and modify constants (`const VERSION = '1.7.8.0';` and `const MINOR_VERSION = 8;` ) and check http://i18n.prestashop.com/translations/1.7.8.0/ku-IQ/ku-IQ.zip is available<br>1. BO > International > Translations > Add / Update a language <br> 2. Add Kurdish language<br>3. :tada: The Kurdish has been installed :tada:

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17826)

### CODE REVIEW

**Comment by LouiseBonnard** [comment]
Hi @SalarNazar, congratulations on your first pull request here! I can read in the issue that a second file ( app/Resources/legacy-to-standard-locales.json ) must be modified to have the Kurdish fully available in PrestaShop. Could you do it as well? It'd be awesome. :-)

**Comment by LouiseBonnard** [comment]
Hi @SalarNazar, congratulations on your first pull request here! I can read in the issue that a second file ( app/Resources/legacy-to-standard-locales.json ) must be modified to have the Kurdish fully available in PrestaShop. Could you do it as well? It'd be awesome. :-)

**Comment by LouiseBonnard** [comment]
Hi @SalarNazar, congratulations on your first pull request here! I can read in the issue that a second file ( app/Resources/legacy-to-standard-locales.json ) must be modified to have the Kurdish fully available in PrestaShop. Could you do it as well? It'd be awesome. :-)

**Comment by Progi1984** [comment]
@LouiseBonnard : @SalarNazar Has done a second PR with corresponding file #17832

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

File: app/Resources/legacy-to-standard-locales.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "an": "an-AR",
+  "kr": "kr-IQ",
```
Comment (LouiseBonnard) [inline app/Resources/legacy-to-standard-locales.json]:
```suggestion
  "ku": "ku-IQ",
```

File: app/Resources/all_languages.json [hunk 2]
```diff
@@ -684,5 +684,14 @@
     "is_rtl": "0",
     "language_code": "zh-cn",
     "locale": "zh-CN"
+  },
+  "kr": {
```
Comment (LouiseBonnard) [inline app/Resources/all_languages.json]:
```suggestion
  "ku": {
```

File: app/Resources/all_languages.json [hunk 3]
```diff
@@ -684,5 +684,14 @@
...
+    "name": "كوردی",
+    "iso_code": "kr",
```
Comment (LouiseBonnard) [inline app/Resources/all_languages.json]:
```suggestion
    "iso_code": "ku",
```

File: app/Resources/all_languages.json [hunk 4]
```diff
@@ -684,5 +684,14 @@
     "is_rtl": "0",
     "language_code": "zh-cn",
     "locale": "zh-CN"
+  },
+  "kr": {
+    "name": "كوردی",
```
Comment (LouiseBonnard) [inline app/Resources/all_languages.json]:
```suggestion
    "name": "Kurdish (کوردی)",
```

File: app/Resources/all_languages.json [hunk 5]
```diff
@@ -684,5 +684,14 @@
...
+    "iso_code": "kr",
+    "date_format_lite": "Y-m-d",
+    "date_format_full": "Y-m-d H:i:s",
+    "is_rtl": "1",
+    "language_code": "kr-iq",
```
Comment (LouiseBonnard) [inline app/Resources/all_languages.json]:
```suggestion
    "language_code": "ku-iq",
```

File: app/Resources/all_languages.json [hunk 6]
```diff
@@ -684,5 +684,14 @@
...
+    "locale": "kr-IQ"
```
Comment (LouiseBonnard) [inline app/Resources/all_languages.json]:
```suggestion
    "locale": "ku-IQ"
```

File: app/Resources/all_languages.json (hunk 4 above)
Comment (LouiseBonnard) [reply app/Resources/all_languages.json]: @SalarNazar, one more commit here and we're done. :-)

**Review by Progi1984 — DISMISSED** [review]

File: app/Resources/all_languages.json [hunk 7]
```diff
@@ -684,5 +684,14 @@
     "is_rtl": "0",
     "language_code": "zh-cn",
     "locale": "zh-CN"
+  },
+  "ku": {
+    "name": "Kurdish (کوردی)",
```
Comment (LouiseBonnard) [inline app/Resources/all_languages.json]:
```suggestion
    "name": "کوردی (Kurdish)",
```

Comment (LouiseBonnard) [reply app/Resources/all_languages.json]: My bad, the English name is in second place. ;-)

**Review by Progi1984 — APPROVED** [review]

**Review by LouiseBonnard — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
Hi @SalarNazar  I don't find kurdish in localization pack or in language can you check this ?

**Comment by SD1982** [comment]
Hi @SalarNazar 
It seems there is no zip file at the provided url 
Screenshot at 10-39-32

And I have an exception while trying to add the kurdish language (probably related with point  1)
https://drive.google.com/file/d/1LqZEXwgHKvJ9wEAOC9hjJflSGh55zicb/view?usp=sharing

**Comment by Progi1984** [comment]
@SD1982 It's normal. Kurdish is a new language in 1.7.8 and it will be released at the 1.7.8 release.
Ping @LouiseBonnard

**Comment by SD1982** [comment]
Hi @Progi1984 Ok so it's the "normal" behaviour so we can'r validate this PR for now as long as we do not have the possibility to test it correctly.
I will test again this PR when the zip will be available :+1:

**Comment by SimonGrn** [comment]
Since the behavior introduced by this PR is validated (Kurdish language is visible in the list and you're correctly redirected to the correct package URL), I would say this is enough to :heavy_check_mark: this PR.
What do you think @Progi1984 ?
Validating the ZIP import is not in this scope in my opinion...

**Comment by Progi1984** [comment]
@SimonGrn I agree. And the ZIP will be available after 178 release.

**Comment by SimonGrn** [comment]
@SD1982 I'll let you validate or not this PR with the info you got from your test then.

Thanks everybody.

**Comment by SD1982** [comment]
Yes I think this PR can be validated :+1:

**Comment by matks** [comment]
Thank you @SalarNazar  @SD1982 @Progi1984

**Pull request merged**